### PR TITLE
Issue 15688 vtl dynamic endpoint

### PR DESCRIPTION
**Pull request by dsilvam**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/FileVelocityReader.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.liferay.util.StringPool;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+
+import static com.dotcms.rest.api.v1.vtl.VTLResource.VTL_PATH;
+
+public class FileVelocityReader implements VelocityReader {
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/FileVelocityReader.java]: We need to add Javadoc to classes and methods

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/RequestBodyVelocityReader.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.RoleAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.Logger;
+
+import java.io.Reader;
+import java.io.StringReader;
+
+public class RequestBodyVelocityReader implements VelocityReader {
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/RequestBodyVelocityReader.java]: Missing Javadoc.

**Review by jcastro-dotcms — COMMENTED** [review]
We need to start adding Javadoc, specially to new classes.

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/VTLResource.java [hunk 3]
```diff
@@ -343,50 +515,153 @@ private void setUserInSession(final HttpSession session, final User user) {
...
-                VelocityUtil.getEngine().evaluate(context, evalResult, "", targetReader);
-            } catch(MethodInvocationException e) {
-                if(e.getCause() instanceof DotToolException) {
-                    throw (Exception) (e.getCause()).getCause();
-                }
+        try {
+            VelocityUtil.getEngine().evaluate(context, evalResult, "", velocityReader);
+        } catch(MethodInvocationException e) {
+            if(e.getCause() instanceof DotToolException) {
+                throw (Exception) (e.getCause()).getCause();
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/VTLResource.java]: Shouldn't be a Logger.error here?

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/FileVelocityReader.java (hunk 1 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/FileVelocityReader.java]: Done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/RequestBodyVelocityReader.java (hunk 2 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/RequestBodyVelocityReader.java]: Done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/VTLResource.java (hunk 3 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/VTLResource.java]: Done

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**